ansible_rulebook/vault.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from ansible.parsing.dataloader import DataLoader
+from ansible.parsing.vault import (
+    VaultLib,
+    AnsibleVaultError,
+    get_file_vault_secret,
+    VaultSecret,
+    PromptVaultSecret,
+    b_HEADER,
+)
+from ansible.parsing.yaml.objects import AnsibleVaultEncryptedUnicode
+
+from ansible_rulebook.exception import VaultDecryptException
+
+
+class Vault:
+    """Vault class allows for decryption."""
+
+    def __init__(
+        self,
+        passwords: list[dict] = None,
+        password_files: list[str] = None,
+        ask_pass: bool = False,
+    ):
+        vault_secrets = []
+        if ask_pass:
+            secret = PromptVaultSecret(
+                vault_id="default", prompt_formats=["Vault password:"]
+            )
+            secret.load()
+            vault_secrets.append((secret.vault_id, secret))
+        for item in passwords or []:
+            if item["type"] == "VaultPassword":
+                vault_secrets.append(
+                    (item["label"], VaultSecret(item["password"].encode()))
```

Comment:
@bzwei can the label be None, from the UX mocks it looks like the label is an optional field and it could be set to None.

Reply:
yes. it can be None.